File: http/src/main/java/demo/http/HttpRoute.java
package demo.http;


import demo.http.SomeProcessor;
import org.apache.camel.builder.RouteBuilder;

/**
 * todo
 */
public class HttpRoute extends RouteBuilder{
    @Override
    public void configure() throws Exception {
        from("jetty:http://localhost:9090/myapp/myservice").streamCaching().
                log("received message ${body}").
                process(new SomeProcessor()).process(new SomeProcessor());

    }
}

File: http/src/main/scala/demo/http/HttpRoute.scala
package demo.http

import org.apache.camel.scala.dsl.builder.ScalaRouteBuilder
import org.apache.camel.{CamelContext, Exchange}

/**
 * todo
 */
class HttpRoute(ctx: CamelContext) extends ScalaRouteBuilder(ctx){
  
  
    "jms:queue/SomeQ" ==> {
        log("message is ${body}")
        convertBodyTo(classOf[String],"UTF-32" )
        process((e: Exchange) => e.in = e.in[String].toUpperCase + "!")
        process((e: Exchange) => e.getIn.setHeader(Exchange.HTTP_URI,"http://localhost:8111/foo"))
        to ("http:DUMMY")
    }

  "jetty:http://localhost:8111/foo" ==> {
    log("received on http 8111 endpoint ${body}")
  }

  "jetty:http://localhost:8112/foo" ==> {
    log("received on http 8112 endpoint ${body}")
  }

}
